lib/util/http/gitlab.spec.ts

```diff
@@ -46,6 +48,29 @@ describe(getName(), () => {
     expect(trace).toHaveLength(3);
     expect(trace).toMatchSnapshot();
   });
+  it('paginates with GITLAB_IGNORE_REPO_URL set', async () => {
+    process.env.GITLAB_IGNORE_REPO_URL = 'true';
+    setBaseUrl(`${selfHostedUrl}/api/v4/`);
+
+    httpMock
+      .scope(selfHostedUrl)
+      .get('/api/v4/some-url')
+      .reply(200, ['a'], {
+        link: '<https://mycompany.com/gitlab/api/v4/some-url&page=2>; rel="next", <https://mycompany.com/gitlab/api/v4/some-url&page=3>; rel="last"',
```

Comment:
```suggestion
        link: '<https://other.host.com/gitlab/api/v4/some-url&page=2>; rel="next", <https://other.host.com/gitlab/api/v4/some-url&page=3>; rel="last"',
```

You want to test the hostname overide, so you should send back a wrong hostname too.

Reply:
It kind of is already (http vs https) but I guess it would be more obvious if i change the hostname too.